src/daemon/daemon.cpp

```diff
@@ -115,10 +116,28 @@ bool Daemon::activate(const InterfaceConfig& config) {
 
   // Bring up the wireguard interface if not already done.
   if (!wgutils()->interfaceExists()) {
+    // Create the interface.
     if (!wgutils()->addInterface(config)) {
       logger.error() << "Interface creation failed.";
       return false;
     }
+
+    // Bring the interface up.
+    if (supportIPUtils()) {
+      if (!iputils()->addInterfaceIPs(config)) {
+        return false;
+      }
+      if (!iputils()->setMTUAndUp(config)) {
+        return false;
+      }
+    }
+
```

Comment:
Just so I better understand, why move this up?

Reply:
The root cause of the issue here is that on Linux and MacOS we need to wait for the interface to go up before we can add routes to it. In the last iteration of work on the Linux firewall (PR #9257) we added some code to check for the interface going up and then load the routing table, which worked fine when the only thing we were loading was a default route, but now that we are are installing multiple routing table entries it gets a lot more complicated to follow that same pattern.

The easier (and cleaner solution) is to just bring the interface up immediately after creation, and before trying to configure a peer and its routes.